src/include/catalog/default_value.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <string>
+#include <utility>
+
+#include "type/type_id.h"
+
+namespace terrier::catalog {
+/**
+ * Wraps the concept of a default value expression into an object that can be
+ * passed between the catalog and the consumers of these expressions.
+ */
```

Comment:
I would be explicit in the comments that this stores a serialized `AbstractExpression`. 
To explain what a serialized expression is, I would maybe add

```
@see src/include/parser/expression/abstract_expression.h
```